packages/core/index.ts

```diff
@@ -34,64 +34,10 @@ export * from './src/exceptions/OperationNotSupportedException';
 export * from './src/exceptions/RangeException';
 export * from './src/exceptions/RuntimeException';
 
-export * from './src/collections/function/AggregateFunction';
-export * from './src/collections/function/CombineFunction';
-export * from './src/collections/function/IteratorFunction';
-export * from './src/collections/function/ProjectFunction';
-
-export * from './src/collections/attributes/Key';
-export * from './src/collections/attributes/readonly/ReadOnlyAttributeAccessor';
-export * from './src/collections/attributes/mutable/AttributeAccessor';
-
-export * from './src/collections/base/CollectionUtils';
-export * from './src/collections/base/KeyValuePair';
-export * from './src/collections/base/MapIteratorFunction';
-export * from './src/collections/base/NamedPool';
-export * from './src/collections/base/NoSuchItemException';
-export * from './src/collections/base/Sequence';
-export * from './src/collections/base/ToArray';
-
-export * from './src/collections/collection/readonly/ReadOnlyCollection';
-export * from './src/collections/collection/readonly/ReadOnlyCollectionBase';
-export * from './src/collections/collection/readonly/ReadOnlyCollectionImpl';
-
-export * from './src/collections/comparison/IterableEqualsFactoty';
-
-export * from './src/collections/iterable/RangeIterable';
-
-export * from './src/collections/list/readonly/ReadOnlyList';
-export * from './src/collections/list/mutable/List';
-export * from './src/collections/list/mutable/ArrayList';
-export * from './src/collections/list/immutable/ImmutableList';
-export * from './src/collections/list/immutable/ImmutableArrayList';
-
-export * from './src/collections/queue/EmptyQueueException';
-export * from './src/collections/queue/readonly/ReadOnlyQueue';
-export * from './src/collections/queue/mutable/ArrayQueue';
-export * from './src/collections/queue/mutable/Queue';
-
-export * from './src/collections/stack/EmptyStackException';
-export * from './src/collections/stack/readonly/ReadOnlyStack';
-export * from './src/collections/stack/mutable/ArrayStack';
-export * from './src/collections/stack/mutable/Stack';
-
-export * from './src/collections/set/readonly/ReadOnlySet';
-export * from './src/collections/set/mutable/Set';
-export * from './src/collections/set/mutable/ArraySet';
-
-export * from './src/collections/map/readonly/ReadOnlyMap';
-export * from './src/collections/map/mutable/Map';
-export * from './src/collections/map/mutable/LinkedMap';
-
-export * from './src/collections/multivaluemap/readonly/ReadOnlyMultiValueMap';
-export * from './src/collections/multivaluemap/mutable/LinkedMultiValueMap';
-export * from './src/collections/multivaluemap/mutable/MultiValueMap';
-
-export * from './src/collections/specialized/BitSet';
-
-export * from './src/collections/tree/TreeNode';
+export * from '@monument/collections';

```

Comment:
Why did you decide to export collections here.
I think after refactoring the core package will not provide collections anymore.

Reply:
It's for BC compatibility to ensure that current users going to have an easy upgrade path. 
As for circular dependency - it will be removed during further steps:ok_hand: